core/src/main/java/com/linecorp/armeria/client/HttpSessionHandler.java

```diff
@@ -311,12 +315,17 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
             channelPool.connect(channel.remoteAddress(), H1C, sessionPromise);
         } else {
             // Fail all pending responses.
-            failUnfinishedResponses(ClosedSessionException.get());
-
+            final Throwable throwable;
+            if (ctx.channel().hasAttr(NESTED_THROWABLE)) {
+                throwable = ClosedSessionException.get(ctx.channel().attr(NESTED_THROWABLE).get());
```

Comment:
Perhaps we can just use the exception as is rather than wrapping this something?